src/api/Volkszaehler.cpp

```diff
@@ -316,7 +316,7 @@ int vz::api::curl_custom_debug_callback(CURL *curl, curl_infotype type, char *da
 
 	case CURLINFO_SSL_DATA_OUT:
 	case CURLINFO_DATA_OUT:
-		data[size] = 0;
+		//data[size] = 0;
```

Comment:
that highlight a severe problem of the current code.The code must not modify the data passed. 

I suggest to completely rework this function. The curl spec says:
```
This callback receives debug information, as specified in the type argument. This function must return 0. The data pointed to by the char * passed to this function WILL NOT be null-terminated, but will be exactly of the size as told by the size argument.
```
So we should not parse the data itself but simply:
- use the size parameter
- if we do really want to log all data sent we'd have to use a "snprintf" alike function or copy the data interims wise into a real 0-terminated string.

Can you rework this like that?
Your current fix is dangerous as we will treat the data as a 0-terminated string which is not the case.

Reply:
```
               char buffer [size];
                snprintf(buffer,size,"%s",data);
                print((log_level_t)(log_debug + 5), "CURL: Sent %lu bytes.. ", ch->name(),
                          (unsigned long)size);
                print((log_level_t)(log_debug + 5), "CURL: Sent '%s' bytes", ch->name(), buffer);
```
Stimmt das so? Oder muss es Buffergröße  ` char buffer [size+1];` sein?

Reply:
ja, der muss 1 größer sein (für die terminierende 0).

Reply:
und ich würde es nicht direkt auf dem stack anlegen.
kannst c++ std::string nehmen:
std::string buffer(data, size);
und dann buffer.c_str() im print
(was ein wenig schade ist, da da 0-en verloren gehen, obwohl der std::string die verwaltet).
Vermutlich sollten wir SSL_IN und SSL_OUT gar nicht loggen, weil es eh verschlüsselt ist.